File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task30.cpp

#include <iostream>
#include <string>
#include <set>
#include <algorithm>

using namespace std;

string longest(const string& s1, const string& s2) {
    set<char> uniqueChars;
    
    for (char c : s1) {
        if (c >= 'a' && c <= 'z') {
            uniqueChars.insert(c);
        }
    }
    
    for (char c : s2) {
        if (c >= 'a' && c <= 'z') {
            uniqueChars.insert(c);
        }
    }
    
    string result;
    for (char c : uniqueChars) {
        result += c;
    }
    
    return result;
}

int main() {
    // Test case 1
    string a1 = "xyaabbbccccdefww";
    string b1 = "xxxxyyyyabklmopq";
    cout << "Test 1: " << longest(a1, b1) << endl;
    
    // Test case 2
    string a2 = "abcdefghijklmnopqrstuvwxyz";
    string b2 = "abcdefghijklmnopqrstuvwxyz";
    cout << "Test 2: " << longest(a2, b2) << endl;
    
    // Test case 3
    string a3 = "aretheyhere";
    string b3 = "yestheyarehere";
    cout << "Test 3: " << longest(a3, b3) << endl;
    
    // Test case 4
    string a4 = "loopingisfunbutdangerous";
    string b4 = "lessdangerousthancoding";
    cout << "Test 4: " << longest(a4, b4) << endl;
    
    // Test case 5
    string a5 = "";
    string b5 = "abc";
    cout << "Test 5: " << longest(a5, b5) << endl;
    
    return 0;
}

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task30.java

import java.util.*;

public class Task30 {
    public static String longest(String s1, String s2) {
        if (s1 == null || s2 == null) {
            return "";
        }
        
        Set<Character> uniqueChars = new TreeSet<>();
        
        for (char c : s1.toCharArray()) {
            if (c >= 'a' && c <= 'z') {
                uniqueChars.add(c);
            }
        }
        
        for (char c : s2.toCharArray()) {
            if (c >= 'a' && c <= 'z') {
                uniqueChars.add(c);
            }
        }
        
        StringBuilder result = new StringBuilder();
        for (char c : uniqueChars) {
            result.append(c);
        }
        
        return result.toString();
    }
    
    public static void main(String[] args) {
        // Test case 1
        String a1 = "xyaabbbccccdefww";
        String b1 = "xxxxyyyyabklmopq";
        System.out.println("Test 1: " + longest(a1, b1));
        
        // Test case 2
        String a2 = "abcdefghijklmnopqrstuvwxyz";
        String b2 = "abcdefghijklmnopqrstuvwxyz";
        System.out.println("Test 2: " + longest(a2, b2));
        
        // Test case 3
        String a3 = "aretheyhere";
        String b3 = "yestheyarehere";
        System.out.println("Test 3: " + longest(a3, b3));
        
        // Test case 4
        String a4 = "loopingisfunbutdangerous";
        String b4 = "lessdangerousthancoding";
        System.out.println("Test 4: " + longest(a4, b4));
        
        // Test case 5
        String a5 = "";
        String b5 = "abc";
        System.out.println("Test 5: " + longest(a5, b5));
    }
}
